backend/gce.go

```diff
@@ -871,12 +908,60 @@ func buildGCEImageSelector(selectorType string, cfg *config.ProviderConfig) (ima
 }
 
 func (p *gceProvider) buildInstance(ctx gocontext.Context, startAttributes *StartAttributes, imageLink, startupScript string) *compute.Instance {
+	logger := context.LoggerFromContext(ctx).WithField("self", "backend/gce_instance")
+
+	var (
+		zone     *compute.Zone
+		err      error
+		diskType string
+	)
+
+	zone = p.ic.Zone
+	if startAttributes.VMConfig.Zone != "" {
+		zone, err = p.client.Zones.Get(p.projectID, startAttributes.VMConfig.Zone).Context(ctx).Do()
+		if err != nil {
+			logger.Warn(err)
+		}
+		logger.Info(fmt.Sprintf("Setting custom diskType: zones/%s/diskTypes/pd-ssd", zone.Name))
+		diskType = fmt.Sprintf("zones/%s/diskTypes/pd-ssd", zone.Name)
+
+	}
+
+	/* TODO: We should do this only if we're using a custom zone to avoid unnecessary lookups. */
 	var machineType *compute.MachineType
 	switch startAttributes.VMType {
 	case "premium":
-		machineType = p.ic.PremiumMachineType
+		pic, err := p.client.MachineTypes.Get(p.projectID, zone.Name, p.cfg.Get("PREMIUM_MACHINE_TYPE")).Context(ctx).Do()
+		if err != nil {
+			logger.Warn(err)
+		}
+		machineType = pic
+	default:
```

Comment:
With only two cases, I don't think this `switch` block is any better than an `if`, but given the API calls happening in each block I don't think it could be replaced with something like:

``` go
if someCondition {
    doSomething()
    return
}

doOtherThing()
```

or

``` go
value := someDefault

if someCondition {
    value = otherValue
}